utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/UniversalSchemaSupport.scala

```diff
@@ -6,107 +6,47 @@ import io.confluent.kafka.schemaregistry.ParsedSchema
-    case schema: AvroSchema => AvroSinkValueParameter(schema.rawSchema())
-    case schema: AvroSchemaWithJsonPayload => AvroSinkValueParameter(schema.rawSchema())
-    case schema: JsonSchema => JsonSinkValueParameter(schema.rawSchema(), defaultParamName = SinkValueParamName)
-    case _ => throw new UnsupportedSchemaType(schema)
-  }
-
-  def sinkValueEncoder(schema: ParsedSchema, validationMode: ValidationMode): Any => AnyRef = schema match {
-    case schema: AvroSchema => (value: Any) => BestEffortAvroEncoder(validationMode).encodeOrError(value, schema.rawSchema())
-    case schema: AvroSchemaWithJsonPayload => (value: Any) => BestEffortAvroEncoder(validationMode).encodeOrError(value, schema.rawSchema())
-    case schema: JsonSchema => (value: Any) =>
-      new BestEffortJsonSchemaEncoder(ValidationMode.lax) //todo: pass real validation mode, when BestEffortJsonSchemaEncoder supports it
-        .encodeOrError(value, schema.rawSchema())
-    case _ => throw new UnsupportedSchemaType(schema)
+    override val payloadDeserializer: UniversalSchemaPayloadDeserializer = support.payloadDeserializer
+    override def serializerFactory[T]: (ConfluentSchemaRegistryClient, KafkaConfig, Boolean) => Serializer[T] = support.serializerFactory
+    override def messageFormatterFactory: SchemaRegistryClient => Any => Json = support.messageFormatterFactory
+    override def messageReaderFactory: SchemaRegistryClient => (Json, String) => Array[Byte] = support.messageReaderFactory
+    override def typeDefinition: TypingResult = support.typeDefinition
+    override def extractSinkValueParameter(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, SinkValueParameter] = support.extractSinkValueParameter
+    override def sinkValueEncoderFactory: ValidationMode => Any => AnyRef = support.sinkValueEncoderFactory
+    override def rawOutputValidatorFactory(implicit nodeId: NodeId): (TypingResult, ValidationMode) => ValidatedNel[OutputValidatorError, Unit] = support.rawOutputValidatorFactory
   }
+}
 
-  def rawOutputValidator(schema: ParsedSchema)(implicit nodeId: NodeId): (TypingResult, ValidationMode) => ValidatedNel[OutputValidatorError, Unit] = (t: TypingResult, validationMode: ValidationMode) =>
-    schema match {
-      case schema: AvroSchema => new AvroSchemaOutputValidator(validationMode).validateTypingResultToSchema(t, schema.rawSchema())
-      case AvroSchemaWithJsonPayload(avroSchema) => new AvroSchemaOutputValidator(validationMode).validateTypingResultToSchema(t, avroSchema.rawSchema())
-      case schema: JsonSchema => new JsonSchemaOutputValidator(validationMode).validateTypingResultToSchema(t, schema.rawSchema())
-      case _ => throw new UnsupportedSchemaType(schema)
-    }
+trait UniversalSchemaSupport {
+  val payloadDeserializer: UniversalSchemaPayloadDeserializer
+  def serializerFactory[T]: (ConfluentSchemaRegistryClient, KafkaConfig, Boolean) => Serializer[T]
```

Comment:
1. what does boolean mean here? maybe we should have some case class for those params?
2. why this method return function instead of just `def createSerializer(client, config, ...)`?